Arm/Insts/DPSFP/Crypto_aes.lean

```diff
@@ -1,45 +1,233 @@
+      0xFEFCFAF8F6F4F2F0EEECEAE8E6E4E2E0#128, -- 7
+      0xDEDCDAD8D6D4D2D0CECCCAC8C6C4C2C0#128, -- 6
+      0xBEBCBAB8B6B4B2B0AEACAAA8A6A4A2A0#128, -- 5
+      0x9E9C9A98969492908E8C8A8886848280#128, -- 4
+      0x7E7C7A78767472706E6C6A6866646260#128, -- 3
+      0x5E5C5A58565452504E4C4A4846444240#128, -- 2
+      0x3E3C3A38363432302E2C2A2826242220#128, -- 1
+      0x1E1C1A18161412100E0C0A0806040200#128  -- 0
+    ]
+  let lo := b.toNat * 8
+  let hi := lo + 7
+  have h : hi - lo + 1 = 8 := by omega
+  h ▸ extractLsb hi lo $ BitVec.flatten FFmul_02
+
+@[state_simp_rules]
+def FFmul03 (b : BitVec 8) : BitVec 8 :=
+  let FFmul_03 :=
+    --   F E D C B A 9 8 7 6 5 4 3 2 1 0
+    [ 0x1A191C1F16151013020104070E0D080B#128, -- F
+      0x2A292C2F26252023323134373E3D383B#128, -- E
+      0x7A797C7F76757073626164676E6D686B#128, -- D
+      0x4A494C4F46454043525154575E5D585B#128, -- C
+      0xDAD9DCDFD6D5D0D3C2C1C4C7CECDC8CB#128, -- B
+      0xEAE9ECEFE6E5E0E3F2F1F4F7FEFDF8FB#128, -- A
+      0xBAB9BCBFB6B5B0B3A2A1A4A7AEADA8AB#128, -- 9
+      0x8A898C8F86858083929194979E9D989B#128, -- 8
+      0x818287848D8E8B88999A9F9C95969390#128, -- 7
+      0xB1B2B7B4BDBEBBB8A9AAAFACA5A6A3A0#128, -- 6
+      0xE1E2E7E4EDEEEBE8F9FAFFFCF5F6F3F0#128, -- 5
+      0xD1D2D7D4DDDEDBD8C9CACFCCC5C6C3C0#128, -- 4
+      0x414247444D4E4B48595A5F5C55565350#128, -- 3
+      0x717277747D7E7B78696A6F6C65666360#128, -- 2
+      0x212227242D2E2B28393A3F3C35363330#128, -- 1
+      0x111217141D1E1B18090A0F0C05060300#128  -- 0
+    ]
+  let lo := b.toNat * 8
+  let hi := lo + 7
+  have h : hi - lo + 1 = 8 := by omega
+  h ▸ extractLsb hi lo $ BitVec.flatten FFmul_03
+
+def AESMixColumns_aux (c : Nat)
+  (in0 : BitVec 32) (in1 : BitVec 32) (in2 : BitVec 32) (in3 : BitVec 32)
+  (out0 : BitVec 32) (out1 : BitVec 32) (out2 : BitVec 32) (out3 : BitVec 32)
+  : BitVec 32 × BitVec 32 × BitVec 32 × BitVec 32 :=
+  if h₀ : 4 <= c then
+    (out0, out1, out2, out3)
+  else
+    let lo := c * 8
+    let hi := lo + 7
+    have h₁ : hi - lo + 1 = 8 := by omega
+    let in0_byte := h₁ ▸ extractLsb hi lo in0
+    let in1_byte := h₁ ▸ extractLsb hi lo in1
+    let in2_byte := h₁ ▸ extractLsb hi lo in2
+    let in3_byte := h₁ ▸ extractLsb hi lo in3
+    let val0 := h₁.symm ▸ (FFmul02 in0_byte ^^^ FFmul03 in1_byte ^^^ in2_byte ^^^ in3_byte)
+    let out0 := BitVec.partInstall hi lo val0 out0
+    let val1 := h₁.symm ▸ (FFmul02 in1_byte ^^^ FFmul03 in2_byte ^^^ in3_byte ^^^ in0_byte)
+    let out1 := BitVec.partInstall hi lo val1 out1
+    let val2 := h₁.symm ▸ (FFmul02 in2_byte ^^^ FFmul03 in3_byte ^^^ in0_byte ^^^ in1_byte)
+    let out2 := BitVec.partInstall hi lo val2 out2
+    let val3 := h₁.symm ▸ (FFmul02 in3_byte ^^^ FFmul03 in0_byte ^^^ in1_byte ^^^ in2_byte)
+    let out3 := BitVec.partInstall hi lo val3 out3
+    have _ : 3 - c < 4 - c := by omega
+    AESMixColumns_aux (c + 1) in0 in1 in2 in3 out0 out1 out2 out3
+  termination_by (4 - c)
+
+@[state_simp_rules]
```

Comment:
Given the complexity of this function, I wouldn't want it to open up during symbolic simulation automatically. I'd recommend removing `@[state_simp_rules]`.

Reply:
Done.